PYME/Analysis/points/ripleys.py

```diff
@@ -76,7 +76,8 @@ def points_from_mask(mask, sampling, three_d = True, coord_origin=(0,0,0)):
     vx, vy, vz = mask.voxelsize
     x0_m, y0_m, z0_m = mask.origin
     x0_p, y0_p, z0_p = coord_origin
-    stride_x, stride_y, stride_z = [max(1, int(sampling / v)) for v in [vx, vy, vz]]
+    safe_div = lambda x, y: int(x/y) if y > 0 else 0
+    stride_x, stride_y, stride_z = [max(1, safe_div(sampling, v)) for v in [vx, vy, vz]]
```

Comment:
this shouldn't be needed - v==0 is an error, and should cause a failure.  We could put an `assert(np.all(np.array(mask.voxelsize)>0)))` in before assigning vx, vy, and vz

Reply:
I guess `vz` could legitimately be 0 if `three_d=False`

Reply:
Correct. And that's a situation I ran into, hence this fix.

Reply:
but suspect the correct way of handling this would be to test for the special case rather than replacing a division with a `safe_div()` which is going to cause us to silently get garbage results

Reply:
What case would silently give us garbage results?

Reply:
ie:

```python
if (vz < eps) and not three_d:
    vz = 1 #dummy value to prevent div by zero when calculating strides we don't use

assert ((vx>0) and (vy>0) and (vz > 0))
```

where `eps` is something small (because vz is a float we can't test `vz==0`).

Reply:
or 'if not (vz > 0):`

Reply:
OK, that is cleaner. I'm cool with changing to that. Why would `safe_div` silently give us garbage results, though?

Reply:
The garbage results case would be when someone loads an image to use as a mask and neglects to define the pixel size. Examples:

- using a different modality (e.g. widefield/confocal/STED) to define the mask
- manually manipulating the image to generate a mask in the shell / an IPython notebook and not keeping track of metadata etc ... 
- (maybe) 3D comparison to 2D mask
- any bug which causes us to drop metadata

I guess my gut feeling before thinking about the 2D case was that something else must have gone wrong to get us to this point.

Reply:
I think if the voxelsize was undefined, `save_div` would still fail, and we would not get garbage results.

```
>>> None > 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

I do think your proposed solution is cleaner, since it directly addresses the `vz==0` case in 2d, which is what I wanted to fix. I've pushed a commit changing to this method of testing.

Reply:
'Undefined' for the voxelsize defaults to either 0 or -1 not `None` (depending on how things are loaded)